src/test/munit/headers/deserializing/header-array-deserializing-with-delimiter-test-suite.xml

```diff
@@ -0,0 +1,1190 @@
+        </munit:enable-flow-sources>
+        <munit:execution>
+            <http:request method="GET" config-ref="http-requester-simple"
+                          path="/headers/validation/test-array-datetime2616-header">
+                <http:headers>#[
+                    output application/json
+                    ---
+                    {
+                    'array-header': 'Sun, 28 Feb 2016 16:41:41 GMT',
+                    'array-header': 'Tue, 15 11 1994 12:45:26 GMT'
+                    }]
+                </http:headers>
+                <http:response-validator>
+                    <http:success-status-code-validator values="1..500"/>
+                </http:response-validator>
+            </http:request>
+        </munit:execution>
+        <munit:validation>
+            <munit-tools:assert-that expression="#[attributes.statusCode]" is="#[MunitTools::equalTo(400)]"/>
+            <munit-tools:assert-that expression="#[payload.description]"
+                                     is="#[MunitTools::containsString('Invalid value')]"/>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="fail-with-invalid-delimited-datetime2616-array-header">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="headers-deserializing-validation-main"/>
+            <munit:enable-flow-source value="get:\test-array-datetime2616-header"/>
+        </munit:enable-flow-sources>
+        <munit:execution>
+            <set-variable variableName="delimiter" value="${delimiter.value}"/>
+            <http:request method="GET" config-ref="http-requester-simple"
+                          path="/headers/validation/test-array-datetime2616-header">
+                <http:headers>#[
+                    %dw 2.0
+                    output application/json
+                    fun setDelimiter(text) = text as String default '' replace "##delimiter##" with vars.delimiter
+                    ---
+                    read(setDelimiter(
+                    "{
+                    'array-header': '1985-04-12T23:20:50.52Z##delimiter##19961219T16:39:57-08:00'
+                    }"
+                    ))
+                    ]
+                </http:headers>
+                <http:response-validator>
+                    <http:success-status-code-validator values="1..500"/>
+                </http:response-validator>
+            </http:request>
+        </munit:execution>
+        <munit:validation>
+            <munit-tools:assert-that expression="#[attributes.statusCode]" is="#[MunitTools::equalTo(400)]"/>
+            <munit-tools:assert-that expression="#[payload.description]"
+                                     is="#[MunitTools::containsString('Invalid value')]"/>
+        </munit:validation>
+    </munit:test>
+
+    <!-- This ignored test can be used in a future for testing different delimiter options (e.g. semicolon) -->
```

Comment:
remove this ?

Reply:
would you like me to remove the tests or the comments? I left them just in case, but not sure if they are going to be used in the future